app/jobs/invoices/prepaid_credit_job.rb

```diff
@@ -7,7 +7,8 @@ class PrepaidCreditJob < ApplicationJob
     unique :until_executed, on_conflict: :log
 
     def perform(invoice)
-      Wallets::ApplyPaidCreditsService.new.call(invoice)
+      wallet_transaction = invoice.fees.find_by(fee_type: 'credit')&.invoiceable
+      Wallets::ApplyPaidCreditsService.new.call(wallet_transaction)
```

Comment:
```suggestion
      Wallets::ApplyPaidCreditsService.call(wallet_transaction)
```

Reply:
I had to reverse this commit because the service accepts the argument in the `call` method. If you want to be able to call it this way, you must accept the argument in the `initialize`

Reply:
I refactored the service in the end ^^